network/kademlia.go

```diff
@@ -902,6 +903,68 @@ func (k *Kademlia) string() string {
 	return "\n" + strings.Join(rows, "\n")
 }
 
+func (k *Kademlia) MarshalJSON() (data []byte, err error) {
+	var (
+		kademliaInfo   = make(map[string]string)
+		depth          = depthForPot(k.conns, k.NeighbourhoodSize, k.base)
+		numConns       = k.conns.Size()
+		numAddrs       = k.addrs.Size()
+		connectedAddrs = make(map[string]struct{})
+		connBinMap     = make(map[string]string)
+		addrsBinMap    = make(map[string]string)
+	)
+
+	jsonizeAddrs := func(m map[string]string) func(bin *pot.Bin) bool {
+		return func(bin *pot.Bin) bool {
+			binMap := make(map[string]string)
+
+			po := bin.ProximityOrder
+			if po >= k.MaxProxDisplay {
+				po = k.MaxProxDisplay - 1 // why???
+			}
+			size := bin.Size
+			poStr := fmt.Sprintf("%d", po)
+			binMap["po"] = poStr
+			binMap["size"] = fmt.Sprintf("%d", size)
+
+			var addresses []string
+			bin.ValIterator(func(val pot.Val) bool {
+				e := val.(*Peer)
+				connectedAddrs[hex.EncodeToString(e.Address())] = struct{}{}
+				addresses = append(addresses, Label(e))
+				return true
+			})
+
+			binMap["addresses"] = strings.Join(addresses, ",")
+			var binString []byte
+			binString, err = json.Marshal(binMap)
```

Comment:
Why marshalling binMap map and assigning it as the string to m map? This will produce an escaped string in the final marshalling, instead a json structure.